src/components/CodeManagement/index.jsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import Helmet from 'react-helmet';
+import { Button, Icon, Pagination } from '@edx/paragon';
+
+import H2 from '../H2';
+import Hero from '../Hero';
+import Coupon from '../Coupon';
+import LoadingMessage from '../LoadingMessage';
+import StatusAlert from '../StatusAlert';
+
+class CodeManagement extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.couponRefs = [];
+    this.state = {
+      hasRequestedCodes: false,
+    };
+
+    this.handleCouponExpand = this.handleCouponExpand.bind(this);
+  }
+
+  componentDidMount() {
+    const { enterpriseId, enterpriseSlug } = this.props;
+
+    if (enterpriseId) {
+      this.props.fetchCouponOrders();
+    }
+
+    this.props.fetchPortalConfiguration(enterpriseSlug);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { enterpriseId, location, history } = this.props;
+
+    if (enterpriseId && enterpriseId !== prevProps.enterpriseId) {
+      this.props.fetchCouponOrders();
+    }
+
+    if (location.state && location.state.hasRequestedCodes) {
+      this.setState({ // eslint-disable-line react/no-did-update-set-state
+        hasRequestedCodes:
```

Comment:
nit: newline here makes this hard to read

Reply:
Already fixed :)